File: algorithms/03_search/linear_search/linear_search.py


def linear_search(n : int , v : int):
    a = list(range(1,n+1))
    exist = False

    for i in a:
        if i == v:
            exist = True
            break

    if exist:
        print("Yes")
    else:
        print("No")


def main():
    linear_search(20,3)


if __name__ == "__main__":
    main()

File: algorithms/03_search/linear_search/linear_search.rs

use std::io;

fn main() {
    // handle input
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let mut parts = input.trim().split_whitespace();
    let n: usize = parts.next().unwrap().parse().unwrap();
    let v: i32 = parts.next().unwrap().parse().unwrap();

    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let a: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    // linear search 
    let exist = a.iter().any(|&x| x == v);

    
    if exist {
        println!("Yes");
    } else {
        println!("No");
    }
}
